src/Symfony/Bridge/Doctrine/IdGenerator/UuidGenerator.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\IdGenerator;
+
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\Id\AbstractIdGenerator;
+use Symfony\Component\Uid\Factory\UuidFactory;
+use Symfony\Component\Uid\Uuid;
+
+final class UuidGenerator extends AbstractIdGenerator
+{
+    private $protoFactory;
+    private $factory;
+    private $entityGetter;
+
+    public function __construct(UuidFactory $factory = null)
+    {
+        $this->protoFactory = $this->factory = $factory ?? new UuidFactory();
+    }
+
+    public function generate(EntityManager $em, $entity): Uuid
+    {
+        if (null !== $this->entityGetter) {
+            if (method_exists($entity, $this->entityGetter)) {
+                return $this->factory->create($entity->{$this->entityGetter}());
+            }
+
+            return $this->factory->create($entity->{$this->entityGetter});
+        }
+
+        return $this->factory->create();
+    }
+
+    /**
+     * @param Uuid|string|null $node
+     *
+     * @return static
+     */
+    public function nameBased(string $entityGetter, $namespace = null): self
```

Comment:
Once this PR is merged, I plan to add support for a new entity annotation that would allow using this method and that would replace the GeneratedValue+CustomIdGenerator combo:

This annotation:
`@ServiceGeneratedValue('uuid.factory', 'nameBased', 'getId', 'some-UUID-namespace')`

would allow populating a property by generating a name-based UUID that would hash the return-value of `$entity->getId()` with `some-UUID-namespace`.